supplementary/alica_ros_turtlesim/src/base.cpp

```diff
@@ -43,6 +47,7 @@ Base::~Base()
     ac->terminate();
     delete ac;
     ALICATurtleWorldModel::del();
+    spinner.stop();
```

Comment:
@veerajsk  looks like spinner::stop is a blocking call http://docs.ros.org/en/melodic/api/roscpp/html/spinner_8cpp_source.html

The way we are doing now could be a problem: AlicaRosCommunication::stop would just set the flag to true and it will stop processing future ros callbacks. But any of spinner thread could already be executing a communication callback. AlicaContext->terminate can then go on shutting down other modules which later get deleted. Can we make the  Scheduler not wait for pending scheduled events to run in case of an engine shutdown?